src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRunner.cs

```diff
@@ -48,6 +49,8 @@ public static async Task<IReadOnlyList<RestoreSummary>> RunAsync(RestoreArgs res
             RestoreArgs restoreArgs,
             CancellationToken token)
         {
+            X509TrustStore.Initialize(restoreArgs.Log);
```

Comment:
Given that the extra certificate bundle is something provided by the SDK, wouldn't it best that it's initialized in SDK code rather than restore code?

In particular, Program.cs in NuGet.Commandline.Xplat, and RestoreTask/PackTask should capture all scenarios.